src/index.js

```diff
@@ -1 +1,32 @@
-export default () => true
+import defaultsDeep from 'lodash.defaultsdeep';
+import pickBy from 'lodash.pickby';
+
+class Stipulate {
+
+  constructor(options) {
+    this.baseOptions = options;
+  }
+
+  prefix(request) {
```

Comment:
potentially, but I'd like the pre and post methods to have name symmetry if possible, and `afterSend` is problematic as mentioned above. gonna return to this when we figure that one out